contracts/PetPark.sol

```diff
@@ -3,5 +3,58 @@ pragma solidity ^0.8.0;
 
 
 contract PetPark {
+    enum AnimalType{None,Fish,Cat,Dog,Rabbit,Parrot}
+    struct PetOwnerDetails{
+        AnimalType pet;
+        uint age;
+        Gender sex;
+    }
+    mapping(address => PetOwnerDetails)petOwner;
+    mapping(AnimalType => uint)public animalCounts;
+
+    enum Gender{Male,Female}
+    address owner;
+    event Added(AnimalType addedAnimal,uint count);
+    event Borrowed(AnimalType borrowedAnimal);
+    constructor(){
+        owner = msg.sender;
+    }
+    function add(AnimalType _animal, uint _count)public {
+        require(owner == msg.sender,"Not an owner");
+        require(_animal != AnimalType.None,"Invalid animal");
+        animalCounts[_animal]+=_count;
+        emit Added(_animal, _count);
+    }
+
+    function borrow(uint age,Gender _gender, AnimalType _animal) public{
+        require(age !=0,"Invalid Age");
+        require(_animal != AnimalType.None,"Invalid animal type");
+        require(animalCounts[_animal] !=0,"Selected animal not available");
+        if(petOwner[msg.sender].age !=0){
+        require(petOwner[msg.sender].age == age,"Invalid Age");
```

Comment:
Since `petOwner[msg.sender]` is being used multiple times, it is better to store this in a separate memory variable. Later in the course, you will learn that this optimised gas

Reply:
Thanks!! looking forward to the lecture.